backtracking/rat_in_maze.py

```diff
@@ -84,35 +158,41 @@ def run_maze(maze: list[list[int]], i: int, j: int, solutions: list[list[int]])
     """
     size = len(maze)
     # Final check point.
-    if i == j == (size - 1):
-        solutions[i][j] = 1
+    if i == destination_row and j == destination_column and maze[i][j] == 0:
+        solutions[i][j] = 0
         return True
 
     lower_flag = (not i < 0) and (not j < 0)  # Check lower bounds
     upper_flag = (i < size) and (j < size)  # Check upper bounds
 
     if lower_flag and upper_flag:
         # check for already visited and block points.
-        block_flag = (not solutions[i][j]) and (not maze[i][j])
+        block_flag = (solutions[i][j]) and (not maze[i][j])
         if block_flag:
             # check visited
-            solutions[i][j] = 1
+            solutions[i][j] = 0
 
             # check for directions
             if (
-                run_maze(maze, i + 1, j, solutions)
-                or run_maze(maze, i, j + 1, solutions)
-                or run_maze(maze, i - 1, j, solutions)
-                or run_maze(maze, i, j - 1, solutions)
+                run_maze(maze, i + 1, j, destination_row, destination_column, solutions)
+                or run_maze(
+                    maze, i, j + 1, destination_row, destination_column, solutions
+                )
+                or run_maze(
+                    maze, i - 1, j, destination_row, destination_column, solutions
+                )
+                or run_maze(
+                    maze, i, j - 1, destination_row, destination_column, solutions
+                )
             ):
                 return True
 
-            solutions[i][j] = 0
+            solutions[i][j] = 1
             return False
     return False
 
 
 if __name__ == "__main__":
     import doctest
 
-    doctest.testmod()
+    doctest.testmod(optionflags=doctest.NORMALIZE_WHITESPACE)
```

Comment:
Does this work?!?

Reply:
> Does this work?!?

Yes, it's working on my local machine, but it's not working on the build test. That's why I added '# doctest: +NORMALIZE_WHITESPACE' to each test.